### PR TITLE
Remove integration tests that require containers

### PR DESCRIPTION
**Pull request by mirekdlugosz**
This brings back integration tests into a pipeline, and removes tests that require containers to run.

Removed tests are ported to Camayoc in https://github.com/quipucords/camayoc/pull/495 .

### CODE REVIEW

**Review by bruno-fs — COMMENTED** [review]
This is good but incomplete. The following files/snippets can be removed (up to you if we do it in this PR or in a follow up):
- `quipucords/tests/integration/conftest.py` can probably be entirely removed
- https://github.com/quipucords/quipucords/blob/2fb0625ceed0b09062a1ca49db9d52057b06411e/pyproject.toml#L53
- probably `quipucords/tests/utils/http.py`
- https://github.com/quipucords/quipucords/blob/bc11757c4c4f259228682faddcd04b38be5202a7/conftest.py#L230-L231

The only thing I ask to be changed in this PR is rewording the commit message titles to follow [conventional commits style](https://www.conventionalcommits.org/en/v1.0.0/).

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -38,6 +38,7 @@ jobs:
             lint,
             check-db-migrations-needed,
             test-coverage,
+            test-integration,
```
[the remark this answers is not in the document]
Comment (mirekdlugosz) [reply .github/workflows/test.yml:41]: You're right, all integration `*Scan` tests (TestAnsibleScan etc) are covered by first make call quoted, and `test_inspect_callback_with_inspect_playbook` is covered by last make call. And these are all tests picked up by `test-integration`.

I will drop the revert commit.

**Comment by mirekdlugosz** [comment]
@bruno-fs All fair points. Thanks!

I included all these changes in 030c158300fc04797893c80fc34e291815621108. It also contains removal of tests from `quipucords/tests/integration/test_network_scan.py` (this part is unchanged).

**Review by bruno-fs — APPROVED** [review]

**Review by infinitewarp — APPROVED** [review]
:shipit:

**Review by ruda — APPROVED** [review]

**Pull request merged**